File: kelmah-mobile-android/app/src/main/java/com/kelmah/mobile/core/security/PasswordPolicy.kt
package com.kelmah.mobile.core.security

object PasswordPolicy {
    const val minimumLength: Int = 8
    const val maximumLength: Int = 128
    const val requirementMessage: String = "Password must be 8-128 characters and include one uppercase letter, one lowercase letter, one number, and one special character"

    fun isStrong(value: String): Boolean =
        value.length in minimumLength..maximumLength &&
            value.any(Char::isUpperCase) &&
            value.any(Char::isLowerCase) &&
            value.any(Char::isDigit) &&
            value.any { !it.isLetterOrDigit() }
}

File: kelmah-mobile-ios/Kelmah/Core/Security/PasswordPolicy.swift
import Foundation

enum PasswordPolicy {
    static let minimumLength = 8
    static let maximumLength = 128
    static let requirementMessage = "Password must be 8-128 characters and include one uppercase letter, one lowercase letter, one number, and one special character"

    static func isStrong(_ value: String) -> Bool {
        value.count >= minimumLength &&
        value.count <= maximumLength &&
        value.rangeOfCharacter(from: .uppercaseLetters) != nil &&
        value.rangeOfCharacter(from: .lowercaseLetters) != nil &&
        value.rangeOfCharacter(from: .decimalDigits) != nil &&
        value.rangeOfCharacter(from: CharacterSet.alphanumerics.inverted) != nil
    }
}